homeassistant/components/climate/spider.py

```diff
@@ -0,0 +1,149 @@
+
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
+    try:
+        api = IthoDaalderop_API(username, password)
+
+    except UnauthorizedException:
+        _LOGGER.exception("Can't connect to the Itho Daalderop API")
+        return
+
+    thermostats = api.get_thermostats()
+    for thermostat in thermostats:
+        add_devices(
+            [SpiderThermostat(api, thermostat)],
+            True
+        )
+
+
+class SpiderThermostat(ClimateDevice):
+    """Representation of a thermostat."""
+
+    def __init__(self, client, thermostat):
+        """Initialize the thermostat."""
+        self.client = client
+        self.thermostat = thermostat
+        self._current_temperature = None
+        self._target_temperature = None
+        self._min_temp = None
+        self._max_temp = None
+        self._id = thermostat['id']
+        self._name = thermostat['name']
+        self._operation = STATE_IDLE
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        supported = (SUPPORT_TARGET_TEMPERATURE)
+
+        return supported
+
+    @property
+    def name(self):
+        """Return the name of the thermostat, if any."""
+        return self._name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return self._min_temp
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return self._max_temp
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is None:
+            return
+
+        self._target_temperature = temperature
+        self.client.set_temperature(self.thermostat, temperature)
+
+    @property
+    def current_operation(self: ClimateDevice) -> str:
+        """Return current operation ie. heat, cool, idle."""
+        return self._operation
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """Get the latest data."""
+        try:
+            for thermostat in self.client.get_thermostats():
+                if thermostat['id'] == self._id:
+                    self.thermostat = thermostat
+
+        except KeyError:
+            _LOGGER.error("Update failed from Itho Daalderop API")
+            return
+
+        except StopIteration:
+            _LOGGER.error("No data from the Itho Daalderop API")
+            return
+
+        for prop in self.thermostat['properties']:
```

Comment:
I wonder if I should set all the properties here or that I should remove the properties and put the logic in the get methods.